### PR TITLE
Enable noImplicitThis TypeScript compiler option 

### PR DESCRIPTION
**Pull request by mohsen1**

### CODE REVIEW

File: lib/Chunk.js [hunk 1]
```diff
@@ -126,7 +127,7 @@ class Chunk {
 
 		/** @private */
 		this._groups = new SortableSet(undefined, sortChunkGroupById);
-		/** @private @type {SortableSet<ChunkGroup>} */
+		/** @private @type {SortableSet<ChunkGroup | Entrypoint>} */
```
Comment (sokra) [inline lib/Chunk.js]: Entrypoint extends ChunkGroup. This should not be needed.

File: lib/Chunk.js [hunk 2]
```diff
@@ -185,7 +186,11 @@ class Chunk {
 	hasRuntime() {
 		for (const chunkGroup of this._groups) {
 			// We only need to check the first one
-			return chunkGroup.isInitial() && chunkGroup.getRuntimeChunk() === this;
+			return (
+				chunkGroup.isInitial() &&
+				// prettier-ignore
+				/** @type {Entrypoint} */ (chunkGroup).getRuntimeChunk() === this
```
Comment (sokra) [inline lib/Chunk.js]: Add a `instanceof Entrypoint` check instead of casting.

Comment (sandersn) [reply lib/Chunk.js]: Alternatively, if Entrypoint.isInitial is equivalent to `instanceof Entrypoint`, then changing the return type of Entrypoint.isInitial to `@returns {this is Entrypoint}` would accomplish the same thing.

Comment (mohsen1) [reply lib/Chunk.js]: Using `Instanceof` will require importing `Entrypoint` which is not optimal. If that's okay, is this what you're suggesting?

```js
return (
    chunkGroup.isInitial() &&
    chunkGroup instanceof Entrypoint &&
    chunkGroup.getRuntimeChunk() === this
);
```

@sandersn per my experience `this is Type` does not work with JSDoc.

File: lib/Chunk.js (hunk 1 above)
Comment (mohsen1) [reply lib/Chunk.js]: Actually we can drop the `Entrypoint`.

File: lib/Chunk.js (hunk 2 above)
Comment (mohsen1) [reply lib/Chunk.js]: Just did `chunkGroup instanceof Entrypoint`

**Comment by sokra** [comment]
Thanks

**Pull request merged**